Mathlib/NumberTheory/SelbergSieve.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2024 Arend Mellendijk. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arend Mellendijk
+-/
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.NumberTheory.ArithmeticFunction
+
+/-!
+# The Selberg Sieve
+
+We set up the working assumptions of the Selberg sieve and define the notion of an upper bound sieve
+and show that every upper bound sieve yields an upper bound on the size of the sifted set. We also
+define the Λ² sieve and prove that Λ² sieves are upper bound sieves. We then diagonalise the main
+term of the Λ² sieve.
+
+We mostly follow the treatment outlined by Heath-Brown in the notes to an old graduate course. One
+minor notational difference is that we write $\nu(n)$ in place of $\frac{\omega(n)}{n}$.
+
+## Results
+ * `siftedSum_le_mainSum_errSum_of_UpperBoundSieve` - Every upper bound sieve gives an upper bound
+ on the size of the sifted set in terms of `mainSum` and `errSum`
+
+## Notation
+The `SelbergSieve.Notation` namespace includes common shorthand for the variables included in the
+`SelbergSieve` structure.
+ * `A` for `support`
+ * `𝒜 d` for `multSum d`
```

Comment:
maybe an explanation of `multSum` could appear here? It's only defined on line 163, and the name isn't completely obvious (until you look at the definition)...